x-pack/plugins/infra/public/components/alerting/logs/expression_editor/criterion_preview_chart.tsx

```diff
@@ -0,0 +1,330 @@
+  getChartTheme,
+  yAxisFormatter,
+  NUM_BUCKETS,
+} from '../../shared/criterion_preview_chart/criterion_preview_chart';
+import {
+  LogDocumentCountAlertParams,
+  Criterion,
+  Comparator,
+} from '../../../../../common/alerting/logs/types';
+import { Color, colorTransformer } from '../../../../../common/color_palette';
+import {
+  GetLogAlertsChartPreviewDataAlertParamsSubset,
+  getLogAlertsChartPreviewDataAlertParamsSubsetRT,
+} from '../../../../../common/http_api/log_alerts/';
+import { AlertsContext } from './editor';
+import { useChartPreviewData } from './hooks/use_chart_preview_data';
+import { decodeOrThrow } from '../../../../../common/runtime_types';
+
+const GROUP_LIMIT = 5;
+
+interface Props {
+  alertParams: Partial<LogDocumentCountAlertParams>;
+  context: AlertsContext;
+  chartCriterion: Partial<Criterion>;
+  sourceId: string;
+}
+
+export const CriterionPreview: React.FC<Props> = ({
+  alertParams,
+  context,
+  chartCriterion,
+  sourceId,
+}) => {
+  const chartAlertParams: GetLogAlertsChartPreviewDataAlertParamsSubset | null = useMemo(() => {
+    const { field, comparator, value } = chartCriterion;
+    const criteria = field && comparator && value ? [{ field, comparator, value }] : [];
+    const params = {
+      criteria,
+      timeSize: alertParams.timeSize,
+      timeUnit: alertParams.timeUnit,
+      groupBy: alertParams.groupBy,
+    };
+
+    try {
+      return decodeOrThrow(getLogAlertsChartPreviewDataAlertParamsSubsetRT)(params);
+    } catch (error) {
+      return null;
+    }
+  }, [alertParams.timeSize, alertParams.timeUnit, alertParams.groupBy, chartCriterion]);
+
+  // Check for the existence of properties that are necessary for a meaningful chart.
+  if (chartAlertParams === null || chartAlertParams.criteria.length === 0) return null;
+
+  return (
+    <CriterionPreviewChart
+      buckets={
+        !chartAlertParams.groupBy || chartAlertParams.groupBy.length === 0
+          ? NUM_BUCKETS
+          : NUM_BUCKETS / 4
+      } // Display less data for groups due to space limitations
+      context={context}
+      sourceId={sourceId}
+      threshold={alertParams.count}
+      chartAlertParams={chartAlertParams}
+    />
+  );
+};
+
+interface ChartProps {
+  buckets: number;
+  context: AlertsContext;
+  sourceId: string;
+  threshold?: LogDocumentCountAlertParams['count'];
+  chartAlertParams: GetLogAlertsChartPreviewDataAlertParamsSubset;
+}
+
+const CriterionPreviewChart: React.FC<ChartProps> = ({
+  buckets,
+  context,
+  sourceId,
+  threshold,
+  chartAlertParams,
+}) => {
+  const isDarkMode = context.uiSettings?.get('theme:darkMode') || false;
+
+  const {
+    getChartPreviewData,
+    isLoading,
+    hasError,
+    chartPreviewData: series,
+  } = useChartPreviewData({
+    context,
+    sourceId,
+    alertParams: chartAlertParams,
+    buckets,
+  });
+
+  useMount(() => {
+    getChartPreviewData();
+  });
+
+  useDebounce(
+    () => {
+      getChartPreviewData();
```

Comment:
This causes `getChartPreviewData()` twice when the component first loads.

I understand is here to update the chart when `chartAlertParams` change, right? Maybe it makes sense to remove the `useMount` call (although that will delay a bit the first load)

Reply:
Yeah, fair point. I don't think the 500ms delay is too much of an issue as this is secondary information, and not the primary reason a user will be using the feature. I'll remove the mount call.